Add bookbag role

jkupferer opened the pull request:
##### SUMMARY

Add role to deploy bookbag from AgnosticD user info

##### ISSUE TYPE

- New role Pull Request

##### COMPONENT NAME

bookbag workload to deploy openshift-homeroom bookbag configuration on an OpenShift cluster using agnosticd user info collected from `agnosticd_user_info` calls.

Update to `test-empty-config` config to allow testing of this role.

wkulhanek commented on ansible/roles_ocp_workloads/bookbag/tasks/pre_workload.yaml:
```diff
@@ -0,0 +1,16 @@
+- name: Check for required variables
+  assert:
+    that:
+    - guid is defined
```
Didn’t you teach me the shorthand?
guid | default(“”)| length >0

jkupferer replied:
Yes, but I thought for showing an assertion error it would be useful for the user to see whether the variable was empty string or undefined as separate failed conditions.

wkulhanek commented on ansible/roles_ocp_workloads/bookbag/tasks/remove_workload.yaml:
```diff
@@ -0,0 +1,8 @@
+---
+# Implement your Workload removal tasks here
```
Nothing to remove?

jkupferer replied:
Oops! I missed pushing that commit! Pushed now. Just removing the namespace.

wkulhanek commented on ansible/roles_ocp_workloads/bookbag/defaults/main.yaml:
```diff
@@ -0,0 +1,9 @@
+#bookbag_git_repo: #required
+#bookbag_git_version: #optional
+bookbag_namespace: "{{ guid }}-bookbag"
+#bookbag_openshift_token: #optional
```
I guess these are for non OpenShift configs?

jkupferer replied:
Exactly.

wkulhanek commented on ansible/roles_ocp_workloads/bookbag/tasks/workload.yaml:
```diff
@@ -0,0 +1,68 @@
+
+- name: Process bookbag build template
+  command: >-
+    oc process --local -f {{ (r_bookbag_tmp.path ~ '/build-template.yaml') | quote }} -o json
```
Is there no way of doing this with k8s and yaml templates?

jkupferer replied:
It could be done with yaml templates, but the way we had folks do bookbag development was using OpenShift templates we gave them for build and deploy. I did it this way to match what the development process.

wkulhanek commented on ansible/roles_ocp_workloads/bookbag/tasks/workload.yaml:
```diff
@@ -0,0 +1,68 @@
+
+- name: Process bookbag deploy template
+  command: >-
+    oc process --local -f {{ (r_bookbag_tmp.path ~ '/deploy-template.yaml') | quote }} -o json
```
Same. No way to do k8s and yaml manifests?

jkupferer commented:
@wkulhanek 

I have added the meta/main.yaml and moved this role to the "roles" directory. Still not 100% on the what is or is not a workload. This one is maybe in-between?

jkupferer commented:
@wkulhanek or @fridim Can I get another review now that this is ready to go? (I think!)

jkupferer commented:
Back to WIP... this should use self-provisioner for namespace creation...
